streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableImpl.java

```diff
@@ -318,7 +477,13 @@ public V apply(Change<V> change) {
         topology.connectProcessorAndStateStores(joinThisName, ((KTableImpl) other).valueGetterSupplier().storeNames());
         topology.connectProcessorAndStateStores(joinOtherName, valueGetterSupplier().storeNames());
 
-        return new KTableImpl<>(topology, joinMergeName, joinMerge, allSourceNodes, null);
+        if (queryableName != null) {
```

Comment:
I wonder if in this case we should always materialize the join table. It would help us with resolving this bug: https://issues.apache.org/jira/browse/KAFKA-4609